resources/design-document.md

```diff
@@ -0,0 +1,223 @@
+
+U8. As a user, I want to be able to update my roles details.
+
+U9. As a user, I want to bb able to view my individual ad(s).
+
+U10. As a user, I want to be able to update my ad's name.
+
+U11. As a user, I want to be able to update my ad's description.
+
+U12. As a user, I want to be able to update my ad's location.
+
+U13. As a user, I want to be able to update my ad's tags.
+
+U14. As a user, I want to be able to update my ad's salary.
+
+U15. As a user, I want to be able to update my ad's name.
+
+U16. As a user, I want to be able to view my user details.
+
+### 3.2  Stretch Use Cases
+
+U1. As a user, I want to be able to view other user's ads.
+
+U2. As a user, I want to be able to apply to another user's ad.
+
+U3. As a user, I want to be able to filter through other user's ads.
+
+U4. As a user, I want to be able to change bulletin boards.
+
+U5. As a user, I want to view venues associated with a board.
+
+## 4. Project Scope
+### 4.1 In Scope
+
+* Creating, retrieving, and updating an ad
+* Retrieving all ads a user has created
+* Updating a user's details. 
+
+### 4.2 Out of Scope
+
+* Multiple Boards (broader than Nashville area) - mid stretch
+* Venues as an entity rather than a String - closest stretch?
+* Allow users to apply to ads and view applications to their ads - farthest stretch 
+
+## 5. Proposed Architecture Overview
+
+The MLP will include creating, retrieving, and updating an ad. Included will also be the functionality to update a user's personal details.
+
+API Gateway and Lambda will be used to create these endpoints to satisfy our requirements:
+* `GetAd`
+* `CreateAd`
+* `UpdateAd`
+* `DeleteAd`
+* `GetUser`
+* `CreateUser`
+* `UpdateUser`
+
+Boards, users, ads, locationsm and venues will be stored within DynamoDB in their own respective tables. 
+
+
+## 6. API
+### 6.1 Public Models
+````
+// UserModel
+
+String userId;
+String name;
+String bio;
+List<String> groups;
+List<String> ads; //adId
+````
+
+````
+// AdModel
+
+String adId;
+String name;
+String description;
+Double salary; //double?
+String location; //venue name
+String userId;
+Set<String> tags;
+````
+
+````
+// LocationModel
+
+String locationId;
+String name;
+List<String> venues; //venueId
+````
+
+````
+// VenueModel
+
+String venueId;
+String name; 
+String description;
+String location;
+````
+
+````
+// BoardModel
+
+String boardId;
+String name;
+Set<String> users; //userIds
+Set<String> ads; //adIds
+String location;
+````
+
+### 6.2 Get Ad Endpoint
+* Accepts `GET` requests to `/ads/:adId`
+
+### 6.3 Create Ad Endpoint
+* Accepts `POST` requests to `/ads`
+
+### 6.4 Update Ad Endpoint
+* Accepts `PUT` requests to `/ads/:adId`
+
+### 6.5 Delete Ad Endpoint
+* Accepts `DELETE` requests to `/ads/:adID`
+
+### 6.6 Get User Endpoint
+* Accepts `GET` requests to `/users/:userId`
+
+### 6.7 Create User Endpoint 
+* Accepts `POST` requests to `/users`
+
+### 6.8 Update User Endpoint
+* Accepts `PUT` requests to `/users/:userId`
+
+## 7. Tables
```

Comment:
Have you thought about where an index might be useful here?

Reply:
I've been working on that decision, and intended to discuss that with you, if we were able to have a 1:1 concerning the overall design of the app.

Reply:
Along with our discussion - I have updated the doc to only include an 'Ads' table rather than both ads and a 'boards' table. The ads table has been more fleshed out and will allow for a GSI using the location or venue. Possibly both.